advanced/metrics.yaml

```diff
@@ -0,0 +1,154 @@
+              - port_rx
+              - frames_tx
+              - frames_rx
+              - bytes_tx
+              - bytes_rx
+              - frames_tx_rate
+              - frames_rx_rate
+              - loss
+              - timestamps
+              - latency
+        metric_group_names:
+          description: >-
+            Extend the advanced flow metrics response by specifying any 
+            configured flow packet header field metric_group names.
+          type: array
+          items:
+            type: string
+          x-constraint:
+            - "/components/schemas/Flow.Pattern/properties/metric_group"
+            
+    AdvancedMetrics.Response:
+      type: array
+      items:
+        $ref: '#/components/schemas/AdvancedMetrics.Flow'
+
+    AdvancedMetrics.Flow:
+      type: object
+      properties:
+        name:
+          description: >-
+            The name of the advanced flow metric response. 
+            The name includes the flow name by default along with any requested 
+            metric_group_names.
+          type: string
+          example: 'Tx -> Rx'
+        transmit:
+          description: >-
+            The transmit state of the flow.
+          type: string
+          enum: [started, stopped, paused]
+        port_tx:
+          description: >-
+            The name of a configured port that the flow is being transmitted 
+            from.
+          type: string
+          x-constraint:
+          - "/components/schemas/Port/properties/name"
+        port_rx:
+          description: >-
+            The name of a configured port that the flow is being recevied on.
+          type: string
+          x-constraint:
+          - "/components/schemas/Port/properties/name"
+        frames_tx:
+          description: >-
+            The current total number of frames transmitted
+          type: integer
+        frames_rx:
+          description: >-
+            The current total number of valid frames received
+          type: integer
+        bytes_tx:
+          description: >-
+            The current total number of bytes transmitted
+          type: integer
+        bytes_rx:
+          description: >-
+            The current total number of bytes received
+          type: integer
+        frames_tx_rate:
+          description: >-
+            The current rate of frames transmitted
+          type: number
+        frames_rx_rate:
+          description: >-
+            The current rate of valid frames received
+          type: number
+        loss:
+          description: >-
+            The percentage of lost frames
+          type: number
+        timestamps:
+          $ref: '#/components/schemas/AdvancedMetrics.Timestamp'
+        latency:
+          $ref: '#/components/schemas/AdvancedMetrics.Latency'
+
+    AdvancedMetrics.Timestamp:
+      description: |-
+        The container for timestamp metrics.
+      type: object
+      properties:
+        first_timestamp_ns:
```

Comment:
can we have timestamp in this format (hours/minutes/seconds) instead of ns?

Reply:
I think we should stick to unix timestamps, otherwise a date-string may further complicate the script.